staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -693,8 +696,26 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 				return nil, err
 			}
 		}
-		// TODO: Once we get rid of /healthz consider changing this to post-start-hook.
-		err := s.addReadyzChecks(healthz.NewInformerSyncHealthz(c.SharedInformerFactory))
+		err = s.AddPostStartHook("generic-apiserver-informers-synced", func(context PostStartHookContext) error {
+			return wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
+				stopCh := make(chan struct{})
+				close(stopCh) // close stopCh to force checking if informers are synced now.
+				informersByStarted := map[bool][]string{}
+
+				for informerType, started := range c.SharedInformerFactory.WaitForCacheSync(stopCh) {
+					switch informerType {
+					// we are interested in knowing if the RBAC related informers sync and avoid the risk that comes with waiting on syncing everything.
+					case reflect.TypeOf(&rbacv1.ClusterRole{}), reflect.TypeOf(&rbacv1.ClusterRoleBinding{}), reflect.TypeOf(&rbacv1.Role{}), reflect.TypeOf(&rbacv1.RoleBinding{}):
```

Comment:
can we depend on RBAC from k8s.io/apiserver? This looks wrong. We only want this in kube-apiserver.

Reply:
what do you mean?

Reply:
The RBAC authorizer is not in this repo, but in k/k/plugins. k8s.io/apiserver is a generic library, while your fix is only for kube-apiserver. In other words, in upstream this would never merge.